src/list-sorting/index.js

```diff
@@ -1,3 +1,29 @@
-function listSorting(needle, haystack) {}
+function listSorting(needle, haystack) {
+    // check if the haystack is a multidimensional array
+    let multidimension = false;
+    for (let element of haystack) {
+      if (typeof element === "object") {
+        multidimension = true;
+      }
+    }
+    
+    // if what we are dealing with is a multidimensional array then we loop 
+    if (multidimension) {
+      const arrayCollector = [];
+      for (let row in haystack) {
```

Comment:
Prefer `for...of` when iterating over an array. `for ... in` is not recommended.